main/src/library/List.flix

```diff
@@ -680,6 +680,18 @@ namespace List {
         case x :: rs => foldRightHelper(f, s, rs, s1 -> k(f(x, s1)))
     }
 
+    ///
+    /// Lazily applies `f` to a start value `s` and all  elements in `xs` going from right to left.
+    ///
+    /// That is, the result is of the form: `f(x1, ...lazy f(xn-1, lazy f(xn, s))...)`.
+    ///
```

Comment:
/// The function `f` must be pure.

/// The foldRightL function exists to allow early termination of a fold.

Reply:
I assume I have to add those since you commented about the extra space, let me know if I need to remove anything